code/modules/surgery/organs/lungs.dm

```diff
@@ -304,24 +286,139 @@
 	// Hydrogen Chloride
 		var/hydrogen_chloride_pp = PP(breath,GAS_HYDROGEN_CHLORIDE)
 		if (prob(hydrogen_chloride_pp))
-			to_chat(H, "<span class='alert'>Your lungs feel terrible!")
+			to_chat(H, span_alert("Your lungs feel terrible!"))
 		if (hydrogen_chloride_pp >20)
 			H.emote("gasp")
-			H.adjustFireLoss(10)
+			H.adjustOxyLoss(10)
 			if (prob(hydrogen_chloride_pp/2))
-				to_chat(H, "<span class='alert'>Your throat closes up!</span>")
+				to_chat(H, span_alert("Your throat closes up!"))
 				H.silent = max(H.silent, 3)
 		else
-			H.adjustFireLoss(round(hydrogen_chloride_pp/4))
+			H.adjustOxyLoss(round(hydrogen_chloride_pp/4))
 		if (gas_breathed > gas_stimulation_min)
 			H.reagents.add_reagent(/datum/reagent/hydrogen_chloride)
 
-	// Stimulum
-		gas_breathed = PP(breath,GAS_STIMULUM)
+	// Carbon Monoxide
+		var/carbon_monoxide_pp = PP(breath,GAS_CO)
+		if (carbon_monoxide_pp > gas_stimulation_min)
+			H.reagents.add_reagent(/datum/reagent/carbon_monoxide, 1)
+			var/datum/reagent/carbon_monoxide/monoxide_reagent = H.reagents.has_reagent(/datum/reagent/carbon_monoxide)
+			if(monoxide_reagent.volume > 10)
+				monoxide_reagent.metabolization_rate = (10 - carbon_monoxide_pp)
+			else
+				monoxide_reagent.metabolization_rate = monoxide_reagent::metabolization_rate
+			switch(carbon_monoxide_pp)
+				if (0 to 20)
+					monoxide_reagent.accumulation = min(monoxide_reagent.accumulation,50)
+				if (20 to 100)
+					monoxide_reagent.accumulation = min(monoxide_reagent.accumulation, 150)
+					H.reagents.add_reagent(/datum/reagent/carbon_monoxide,2)
+				if (100 to 200)
+					monoxide_reagent.accumulation = min(monoxide_reagent.accumulation, 250)
+					H.reagents.add_reagent(/datum/reagent/carbon_monoxide,4)
+				if (200 to 400)
+					monoxide_reagent.accumulation = min(monoxide_reagent.accumulation, 250)
+					H.reagents.add_reagent(/datum/reagent/carbon_monoxide,8)
+				if (400 to INFINITY)
+					monoxide_reagent.accumulation = max(monoxide_reagent.accumulation, 450)
+					H.reagents.add_reagent(/datum/reagent/carbon_monoxide,16)
+		else
+			var/datum/reagent/carbon_monoxide/monoxide_reagent = H.reagents.has_reagent(/datum/reagent/carbon_monoxide)
+			if(monoxide_reagent)
+				monoxide_reagent.accumulation = min(monoxide_reagent.accumulation, 150)
+				monoxide_reagent.metabolization_rate = 10 //purges 10 per tick
+
+	//if you can read this, please tell me a good way to reduce repeating this piece of code
+	// Sulfur Dioxide
```

Comment:
you could probably make this stuff a proc instead of jamming it into the code here. have the fire conditions call the proc.